nmostesting/suites/IS0401Test.py

```diff
@@ -1556,6 +1563,292 @@ def test_26(self, test):
+        no_meta = True
+        all_meta = True
+        if receivers_valid and receivers_response.status_code == 200:
+            try:
+                for receiver in receivers_response.json():
+                    no_receivers = False
+                    if "constraint_sets" in receiver["caps"]:
+                        no_constraint_sets = False
+                        for constraint_set in receiver["caps"]["constraint_sets"]:
+                            if "urn:x-nmos:cap:meta:" + meta in constraint_set:
+                                no_meta = False
+                            else:
+                                all_meta = False
+            except json.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from Node API")
+            except KeyError as e:
+                return test.FAIL("Unable to find expected key in the Receiver: {}".format(e))
+
+        if no_receivers:
+            return test.UNCLEAR("No Receivers were found on the Node")
+        elif no_constraint_sets:
+            return test.OPTIONAL("No BCP-004-01 'constraint_sets' were identified in Receiver caps",
+                                 "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                 "/docs/1.0._Receiver_Capabilities.html#listing-constraint-sets"
+                                 .format(api["spec_branch"]))
+        elif no_meta:
+            return test.OPTIONAL("No BCP-004-01 'constraint_sets' have {}".format(description),
+                                 "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                 "/docs/1.0._Receiver_Capabilities.html#constraint-set-{}"
+                                 .format(api["spec_branch"], meta))
+        elif warn_not_all and not all_meta:
+            return test.WARNING("Only some BCP-004-01 'constraint_sets' have {}".format(description),
+                                "https://amwa-tv.github.io/nmos-receiver-capabilities/branches/{}"
+                                "/docs/1.0._Receiver_Capabilities.html#constraint-set-{}"
+                                .format(api["spec_branch"], meta))
+        else:
+            return test.PASS()
+
+    def test_27_7(self, test):
+        """Receiver 'caps' parameter constraints should be used with the correct format"""
+
+        # general_constraints = [
+        #     "urn:x-nmos:cap:format:media_type",
+        #     "urn:x-nmos:cap:format:grain_rate"
+        # ]
+        video_specific_constraints = [
+            "urn:x-nmos:cap:format:frame_width",
+            "urn:x-nmos:cap:format:frame_height",
+            "urn:x-nmos:cap:format:interlace_mode",
+            "urn:x-nmos:cap:format:colorspace",
+            "urn:x-nmos:cap:format:transfer_characteristic",
+            "urn:x-nmos:cap:format:color_sampling",
+            "urn:x-nmos:cap:format:component_depth",
+            "urn:x-nmos:cap:transport:st2110_21_sender_type"

```

Comment:
🤔 Transport constraint only for video? N, NL, etc apply to all formats?

Reply:
Oh, hmmm, are you sure? The title of [ST 2110-21](https://ieeexplore.ieee.org/document/8165971) is "Traffic Shaping and Delivery Timing for Video".

Reply:
😖 not any more! I usually think of [this talk by leigh](https://theiabm.org/deep-dive-smpte-st-2110-21-traffic-shaping-delivery-timing-video/) but I don't recall any mention of media type... rewatching all the slides say video! 📹 
But I always assumed the rules would be applied to other media types ... 20s are video specs... 🙊 

Never mind me and my confusion, this is spot on! 👍